src/main/java/cstwooneohthree/glados/utils/Ui.java

```diff
@@ -0,0 +1,199 @@
+
+import java.util.ArrayList;
+
+import cstwooneohthree.glados.enums.UiType;
+import cstwooneohthree.glados.tasks.Task;
+
+/**
+ * Ui class to handle printing to command line or returning strings in GUI.
+ *
+ * @author jayjay19630
+ */
+public class Ui {
+
+    /* Dividers and logo to display in CLI */
+    private static final String HORIZONTAL_LINE = "\n"
+            + "-----------------------------------------------------------------------------\n";
+    private static final String LOGO = "\n"
+            + "\n ░▒▓██████▓▒░░▒▓█▓▒░       ░▒▓██████▓▒░░▒▓███████▓▒░ ░▒▓██████▓▒░ ░▒▓███████▓▒░ \n"
+            + "░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░        \n"
+            + "░▒▓█▓▒░      ░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░        \n"
+            + "░▒▓█▓▒▒▓███▓▒░▒▓█▓▒░      ░▒▓████████▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░░▒▓██████▓▒░  \n"
+            + "░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░      ░▒▓█▓▒░ \n"
+            + "░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░      ░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓█▓▒░░▒▓█▓▒░      ░▒▓█▓▒░ \n"
+            + " ░▒▓██████▓▒░░▒▓████████▓▒░▒▓█▓▒░░▒▓█▓▒░▒▓███████▓▒░ ░▒▓██████▓▒░░▒▓███████▓▒░  \n"
+            + "\n";
+    private UiType uiType;
+
+    public Ui(UiType uiType) {
+        this.uiType = uiType;
+    }
+
+    /**
+     * Prints or returns greeting statement.
+     */
+    public String greet() {
+        String message = "\nHello, welcome to the Aperture Science computer-aided enrichment center! My name is: " 
+                         + LOGO + "What task would you like me to perform today?" + "\n";
+
+        if (uiType == UiType.COMMAND_LINE_INTERFACE) {
+            System.out.println(HORIZONTAL_LINE + message + HORIZONTAL_LINE);
+            return null;
+        } else {
+            return "Hello, welcome to the Aperture Science computer-aided enrichment center!"
+                    + "My name is GLaDOS. What task would you like me to perform today?";
+        }
+    }
+
+    /**
+     * Prints or returns user input in echo command.
+     *
+     * @param input Input to be echoed.
+     */
+    public String echo(String input) {
+        String message = "\nGLaDOS: " + input.trim() + "\n";
+
+        if (uiType == UiType.COMMAND_LINE_INTERFACE) {
+            System.out.println(HORIZONTAL_LINE + message + HORIZONTAL_LINE);
+            return null;
+        } else {
+            return input.trim();
+        }
+    }
+
+    /**
+     * Prints or returns result after task has been added.
+     *
+     * @param task Task description to be added.
+     * @param index Number of tasks left.
+     */
+    public String add(String task, String index) {
+        String message = "\nGLaDOS: I have added the following task to the list...\n" 
+                        + task + "\n" 
+                        + "Now you have " + index + (Integer.parseInt(index) == 1 ? " task." : " tasks.")
+                        + "\n";
+
```

Comment:
Good job considering even the most minor stuff, singular for a task and plural for multiple tasks!